### ISSUE TITLE
First Slice Email Integration - Send single complainant form letter from CP

### ISSUE THREAD

**Issue by randyabramson** [usdoj-crt/crt-portal-management#803]
### Objective: 
Reduce the amount of time intake specialists spend on communicating with complainants

### Key Result: 
Implement Email integration that will speed up response time and increase backlog burn rate by TBD %
-	On 11/25/20, backlog is 1,222
-	Need to figure out normal close rate per week
-	Compare with new burn rate

**As an intake specialist, I want to be able to send a single complainant a trending form letter straight from the Complaint Portal**

### Acceptance Criteria

Given that I am on the Single Complaint View
When I select the Contact Complainant button
The I see the Contact Complainant screen

Given that I am on the Contact Complainant screen
When I look at the pre-populated text
Then I see pre-populated AND un-editable fields for the Complainant Full Name AND the Complainant Email address (if available) AND the Response Subject line AND the empty form letter dropdown AND the blank form letter box

Given that I am on the Contact Complainant screen
When I select a form letter
Then I see the buttons that say COPY, PRINT and SEND 

Given that I am on the Contact Complainant screen
When I select a form letter AND there is Email address for this report
Then the buttons that say COPY, PRINT and SEND are BLUE and ACTIVE

Given that I am on the Contact Complainant screen
When I select a form letter
Then the form letter is not editable

Given that I pressed the SEND button
When I return to the individual complaint screen
Then I see a message that says ‘Email sent {form letter name} to [complainant's email address] via {mail service}

Given that I pressed the SEND button
When I return to the individual complaint screen
Then I see a message in the activity log that says 'Email sent {form letter name} to [complainant's email address] via {mail service}'

### Assumptions

- The language 'Response: Your Civil Rights Division Report - [Complaint ID#] from the [Assigned] Section' will be the subject line of the email.
- The required unsubscribe text will automatically be added to the bottom of every email after the CRT User clicks the 'Send' button
- The public user provided a valid email address when submitting their complaint
- If an Email is not included in the report, the send button is 'blue' but not active (this will be fleshed out in a separate card)
- There will be a separate card to address: 
* Unsubscribe features

Error state
- Mail not sent error - Bounce or other failure to send alerts intake specialist and error written to activity log

**Comment by randyabramson** [comment]
Adding mocks task to this ticket for @emilycryan

**Comment by emilycryan** [comment]
Screens and click-thrus!

For the "happy path": https://projects.invisionapp.com/share/4BZKTCT6RAN#/screens

There is one case where we may not have an email address. In this case, we still want the user to be able to see the COPY and PRINT options. In this case, we will keep the SEND button greyed out. For a view of this screen, go to:

https://projects.invisionapp.com/share/4BZKTCT6RAN#/screens/439118013

**Comment by randyabramson** [comment]
Nice work @emilycryan !

**Comment by randyabramson** [comment]
@emilycryan it was requested that we deal with reports without Email addresses in a separate card

**Comment by emilycryan** [comment]
@randyabramson Ok, then let's put the single screen view into whatever the new card is for that story. This is at: https://projects.invisionapp.com/share/4BZKTCT6RAN#/screens/439118013

-e

**Comment by Jkrzy** [comment]
Are the messages we're providing to an intake specialist and recording in the activity log intentionally different?

> Given that I pressed the SEND button
> When I return to the individual complaint screen
> Then I see **a message that says ‘Email sent {form letter name} to [complainant's email address] via {mail service}**
> 
> Given that I pressed the SEND button
> When I return to the individual complaint screen
> *Then I see a message in the **activity log that says ‘Email sent to complainant via {mail service}**

I'd like to use  "Email sent {form letter name} to [complainant's email address] via {mail service}" for both.

**Comment by emilycryan** [comment]
Agree with @Jkrzy They should be consistent. And def include the letter in the activity log.

**Comment by randyabramson** [comment]
Good catch @Jkrzy - they should be the same. I will change the ACs

**Comment by Jkrzy** [comment]
Making progress here: https://github.com/usdoj-crt/crt-portal/pull/793.

We're blocked on sending a real outbound email by AWS SES access and verification for which we have a question out to JMD and a meeting scheduled next week.

**Comment by Jkrzy** [comment]
We're limited in how we communicate "errors" to an intake specialist, specifically there's no immediate feedback available re: the delivery of an outbound message to its' destination. 

Initially, we'll need to assess delivery of emails directly in the AWS SES Console.

**Comment by randyabramson** [comment]
Awesome news, @Jkrzy! Congrats! {Side note, I removed the 'blocked' label from this since we have (limited) access to the sandbox.

**Comment by randyabramson** [comment]
Good point on send error feedback, @Jkrzy. This is where having one box gets a bit sticky. Maybe reviewing this box will have to be a centralized task for the first slice.

**Comment by Jkrzy** [comment]
We'll also need to update our compliance documentation, Lindsay tracked a few items here!
 
https://app.zenhub.com/workspaces/doj-crt-intake-scrum-board-5d03bf56c1c8a35d482eca0f/issues/18f/crt-portal/859

Let's pair on that issue @ilodidoj!

**Comment by ilodidoj** [comment]
@Jkrzy definitely, I can start pairing on Monday next week or time that doable next week.

**Comment by randyabramson** [comment]
@Jkrzy I think I failed to include an AC for what to display if the user supplied and Email but not a first name, last name

One suggestion is to sub in 'Dear friend' for these cases. Thoughts?

cc: @blaughman

**Comment by blaughman** [comment]
@Jkrzy @randyabramson 

Currently, responses to reports without a name start with

"Thank you for your report,"

I think for consistency sake, we should continue this practice.

**Comment by emilycryan** [comment]
Yes, the text should not change at all between the "copy" or "print" text blurb and the amazon SES email text. If a response gets modified, that should only happen by the intake specialist when/if they decide to copy and paste it into an Outlook email and are modifying it for a very specific purpose. 

CC @blaughman @randyabramson

**Comment by randyabramson** [comment]
All -- thanks for the brain power on the above.  Thanks for helping to think it through.

**Comment by Jkrzy** [comment]
This is up for review - https://github.com/usdoj-crt/crt-portal/pull/793/

Added [documentation on configuration here](https://github.com/usdoj-crt/crt-portal/blob/d7698e019997de8d5caec6bb8dffa8da2a531c28/docs/maintenance_or_infrequent_tasks.md#email-configuration).

The final pieces we need are:

- [ ] AWS Credentials (#765)
- [ ] A FROM email address, verified in AWS SES (#851)
- [ ] A ReturnPath email address, a working and monitored inbox to which complaint and hard bounce notifications will be received from AWS SES

**Comment by randyabramson** [comment]
@Jkrzy  thanks for the thorough documentation on the above ^^^

**Comment by randyabramson** [comment]
@Jkrzy @timoballard @ilodidoj @AvivaOskow @blaughman 

Including @emilycryan 'text only' mocks for form letters here.

I am not sure if bolding/underlining fall into the HTML category of letters, but I hope that these will at least help with where to place character returns.

**Comment by randyabramson** [comment]
Just updating here: Imran sent this to Jill on 5/4/2021
Hi Jill:
Just following up to see if you have any latest update on the status or timeline for DKIM binding/pairing from your engineering team. We will be concentrating more on the email integration in current and next sprint.
Thanks.
-	Imran

**Comment by randyabramson** [comment]
@blaughman 

There's an issue with the Webhooks where we can't see any reporting of failed Email sends, bounces, etc.

We can still move forward with the project, but where we end up on reporting is a bit up in the air. Per @Jkrzy, these are our choices (cc: @ilodidoj )

1.	Wait for webhooks to work properly (Jill checking with Richard)
2.	Build in our own reporting
a.	Could be Imran pinging the system each week to tell us which Emails failed
b.	Or, something very robust where it is displayed in the Django console where we currently see Email record activity
